frontend/src/Components/ContentComponent.tsx

```diff
@@ -104,6 +106,66 @@ function updateContent(
   div.querySelectorAll('div.oauth-app').forEach((appEl) => {
     updateOauthAppEmbed(appEl as HTMLDivElement, appState)
   })
+  div.querySelectorAll('div[data-telegram-url]').forEach((telegramDiv) => {
+    updateTelegramEmbed(telegramDiv as HTMLElement, appState)
+  })
+}
+
+function updateTelegramEmbed(telegramDiv: HTMLElement, appState: AppState) {
```

Comment:
It is very much existing `updateInternalExpandButton`. Could you abstract shared functionality into separate function for more clear code please